contributions/admin.py

```diff
@@ -8,12 +8,75 @@
 from django.contrib.admin.decorators import display
 from django.utils.translation import gettext_lazy as _
 from django.utils.translation import ngettext as _p
+from import_export import resources
+from import_export.admin import ExportMixin
+from import_export.fields import Field
 from rangefilter.filters import DateTimeRangeFilter
+import datetime
+import logging
 
 
-class ContributionAdmin(admin.ModelAdmin):
+logger = logging.getLogger('django')
+
+
+class ContributionResource(resources.ModelResource):
+    """Define the export for Contributions."""
+
+    def _f(field, name=None):
+        """Shortcut for field creation."""
+        return Field(attribute=field, column_name=name)
+
+    number = _f('license__number', _('License number'))
+    title = _f('license__title', _('Title'))
+    subtitle = _f('license__subtitle', _('Subtitle'))
+    broadcast_date = _f('broadcast_date__date', _('Broadcast Date'))
+    broadcast_time = _f('broadcast_date__time', _('Broadcast Time'))
+    profile = _f('license__profile', _('Profile'))
+    duration = _f('license__duration', _('Duration'))
+
+    class Meta:
+        """Define meta properties for Contribution export."""
+
+        model = Contribution
+        fields = ['live']
+
+
+class YearFilter(admin.SimpleListFilter):
+    """Filter after this or last years broadcast_date."""
+
+    title = _('Broadcast year')
+    parameter_name = 'broadcast_date'
+
+    def lookups(self, request, model_admin):
+        """Define labels to filter after this or last year."""
+        return (
+            ('this', _('This year')),
+            ('last', _('Last year')),
+        )
+
+    def queryset(self, request, queryset):
+        """Filter after broadcast date for this or last year."""
+        if self.value() is None:
+            return
+
+        match self.value():
+            case 'this':
+                return queryset.filter(
```

Comment:
It is not obvious to me, why the first comparison is not part of the match statement. Is this not possible with `None` values? Same applies to the occurrence in `licenses.admin` later on.

Reply:
Yes, `case None:` should work.

Reply:
[e1141ce](https://github.com/gocept/ok_tools/pull/79/commits/e1141cecc9744414feb38534907534f01a8547d4)